### ISSUE TITLE
Implement systems

### ISSUE THREAD

**Issue by heinezen** [SFTtech/openage#1684]
Overview of the necessary gameplay [systems](https://github.com/SFTtech/openage/blob/master/doc/code/game_simulation/game_entity.md#system) that we need to implement:

- [ ] Move
- [ ] Idle
- [ ] Activity
- [ ] Create
- [ ] Research
- [ ] Trade
- [ ] Exchange Resources
- [ ] Transform
- [ ] Gather
- [ ] Restock
- [ ] Drop resources
- [ ] Herd
- [ ] Despawn
- [ ] Apply effect
- [ ] Shoot projectile
- [ ] Transfer storage
- [ ] Collect storage
- [ ] Remove storage
- [ ] Enter container
- [ ] Exit container
- [ ] Send back to task

**Comment by jere8184** [comment]
Can I start trying to implement some of these systems?

**Comment by heinezen** [comment]
@jere8184 You can but it would  involve basically reading all the documentation and then also designing these systems from the ground up, which is not very easy. Just so you know.

I think the easiest systems might be the enter container/exit container systems.

**Comment by jere8184** [comment]
will the container also be a `GameEntity`? If so I'm guessing it will need some type of component indicating it is enterable and what type of `GameEntity ` It is able to store?

**Comment by heinezen** [comment]
@jere8184 Yes, one game entity has a "container" component, the other game entity has an "enter container" component and then the latter should be able to enter the former.

What type of entities can be stored has to be read from the modpack data via the nyan API, i.e. the component should reference the associated API object (e.g. [`ability.type.Storage`](https://github.com/SFTtech/openage/blob/master/doc/nyan/api_reference/reference_ability.md#abilitytypestorage) for the container definition). When the container is entered, the system function would figure out if the entering game entity (A) can be stored in the entered game entity's (B) container by reading the container definition of B from the modpack data via the nyan API. If that succeeds, the system would then put A into B.
